rusqlite_migration_tests/tests/from_directory_test.rs

```diff
@@ -0,0 +1,138 @@
+use std::num::NonZeroUsize;
+
+use include_dir::{include_dir, Dir};
+
+use rusqlite::{params, Connection};
+use rusqlite_migration::{Error, Migrations, SchemaVersion};
+
+static MIGRATIONS_DIR: Dir =
+    include_dir!("$CARGO_MANIFEST_DIR/../examples/from-directory/migrations");
+static JUST_DOWN: Dir = include_dir!("$CARGO_MANIFEST_DIR/tests/migrations/just_down");
+static EMPTY_DIR: Dir = include_dir!("$CARGO_MANIFEST_DIR/tests/migrations/empty_dir");
+static NON_CONSECUTIVE: Dir = include_dir!("$CARGO_MANIFEST_DIR/tests/migrations/non_consecutive");
+static MULTIPLE: Dir = include_dir!("$CARGO_MANIFEST_DIR/tests/migrations/multiple");
+static ZERO_AS_ID: Dir = include_dir!("$CARGO_MANIFEST_DIR/tests/migrations/zero_as_id");
+static INVALID_UTF8: Dir = include_dir!("$CARGO_MANIFEST_DIR/tests/migrations/invalid_utf8");
+static TOO_LARGE_MIGRATION_ID: Dir =
+    include_dir!("$CARGO_MANIFEST_DIR/tests/migrations/too_large_migration_id");
+static BAD_MIGRATION_ID: Dir =
+    include_dir!("$CARGO_MANIFEST_DIR/tests/migrations/bad_migration_id");
+static MISSING_MIGRATION_ID: Dir =
+    include_dir!("$CARGO_MANIFEST_DIR/tests/migrations/missing_migration_id");
+
+#[test]
+fn main_test() {
+    simple_logging::log_to_stderr(log::LevelFilter::Trace);
+
+    let mut conn = Connection::open_in_memory().unwrap();
+    // Define migrations
+
+    {
+        let migrations = Migrations::from_directory(&MIGRATIONS_DIR).unwrap();
+
+        migrations.to_latest(&mut conn).unwrap();
+
+        assert_eq!(
+            Ok(SchemaVersion::Inside(NonZeroUsize::new(3).unwrap())),
+            migrations.current_version(&conn)
+        );
+
+        conn.execute(
+            "INSERT INTO friend (name, birthday) VALUES (?1, ?2)",
+            params!["John", "1970-01-01"],
+        )
+        .unwrap();
+    }
+}
+
+#[test]
+fn utf8_test() {
+    simple_logging::log_to_stderr(log::LevelFilter::Trace);
```

Comment:
In my opinion yes, since we should verify if the errors are correctly handled. I'm not sure about the duplicate tests though - the only thing they give us is higher coverage.

Reply:
Oh sorry, I did not mean the test, just the logging bit, at trace level. So just this line:
```rust
    simple_logging::log_to_stderr(log::LevelFilter::Trace);
```

Reply:
In reality I don't think this line is necessary. I'll remove it from the tests.